sw/device/tests/sim_dv/pwrmgr_sleep_all_wake_ups_impl.c

```diff
@@ -108,176 +189,278 @@ static void prgm_pinmux_wakeup(void *dif) {
+
   dif_usbdev_phy_pins_drive_t pins = {
       .dp_pullup_en = true,
       .dn_pullup_en = false,
   };
-  CHECK_DIF_OK(dif_usbdev_set_phy_pins_state(dif, kDifToggleEnabled, pins));
-  CHECK_DIF_OK(dif_usbdev_set_wake_enable(dif, kDifToggleEnabled));
+  CHECK_DIF_OK(dif_usbdev_set_phy_pins_state(&dif, kDifToggleEnabled, pins));
+  CHECK_DIF_OK(dif_usbdev_set_wake_enable(&dif, kDifToggleEnabled));
 
   LOG_INFO("prgm_usb_wakeup: wait 20us (usb)");
   // Give the hardware a chance to recognize the wakeup values are the same.
   busy_spin_micros(20);  // 20us
 }
 
+static void prgm_usb_check(dt_pwrmgr_wakeup_src_t src) {
+  // No bit in USBDEV indicates it caused a wakeup.
+}
+
+static void prgm_usb_clear(dt_pwrmgr_wakeup_src_t src) {
+  INIT_DIF_DT(pinmux, src, dt, pinmux)
+  dif_usbdev_t usbdev;
+  static const dt_usbdev_t dt_usbdev = 0;
+  static_assert(kDtUsbdevCount == 1, "expect exactly one usbdev");
+  CHECK_DIF_OK(dif_usbdev_init_from_dt(dt_usbdev, &usbdev));
+
+  CHECK_DIF_OK(dif_usbdev_set_wake_enable(&usbdev, kDifToggleDisabled));
+  // Write again to make sure the first one has already completed.
+  CHECK_DIF_OK(dif_usbdev_set_wake_enable(&usbdev, kDifToggleDisabled));
+  CHECK_DIF_OK(dif_pinmux_wakeup_cause_clear(&pinmux));
+}
+#endif /* HAS_USBDEV */
+
 /**
  * aon timer config for test #5
  * set wakeup signal in 50us
  */
-static void prgm_aontimer_wakeup(void *dif) {
-  CHECK_STATUS_OK(aon_timer_testutils_wakeup_config(dif, 10));
+static void prgm_aontimer_wakeup(dt_pwrmgr_wakeup_src_t src) {
+  INIT_DIF_DT(aon_timer, src, dt, aon_timer)
+  // Enable all the AON interrupts used in this test.
+  dif_rv_plic_irq_id_t irq_id =
+      dt_pwrmgr_irq_to_plic_id(kPwrmgrDt, kDtPwrmgrIrqWakeup);
+  rv_plic_testutils_irq_range_enable(&rv_plic, kPlicTarget, irq_id, irq_id);
+  CHECK_STATUS_OK(aon_timer_testutils_wakeup_config(&aon_timer, 10));
+}
+
+static void prgm_aontimer_check(dt_pwrmgr_wakeup_src_t src) {
+  INIT_DIF_DT(aon_timer, src, dt, dif)
+  bool cause = false;
+  CHECK_DIF_OK(dif_aon_timer_get_wakeup_cause(&dif, &cause));
+  CHECK(cause, "Expected aontimer wakeup cause to be enabled");
+}
+
+static void prgm_aontimer_clear(dt_pwrmgr_wakeup_src_t src) {
+  INIT_DIF_DT(aon_timer, src, dt, aon_timer)
+  CHECK_DIF_OK(dif_aon_timer_wakeup_stop(&aon_timer));
+  // Write again to make sure the first one has already completed.
+  CHECK_DIF_OK(dif_aon_timer_wakeup_stop(&aon_timer));
+  CHECK_DIF_OK(dif_aon_timer_clear_wakeup_cause(&aon_timer));
 }
 
+#ifdef HAS_SENSOR_CTRL
+//   #include "sw/device/lib/dif/dif_sensor_ctrl.h"
+// #include "sensor_ctrl_regs.h"
 /**
  * sensor ctrl config for test #6
  * setup event trigger0
  */
-static void prgm_sensor_ctrl_wakeup(void *dif) {
-  CHECK_DIF_OK(dif_sensor_ctrl_set_ast_event_trigger(dif, kSensorCtrlEventIdx,
-                                                     kDifToggleEnabled));
+static void prgm_sensor_ctrl_wakeup(dt_pwrmgr_wakeup_src_t src) {
```

Comment:
I know it's a pre-existing issue, but please rename these functions to `_config` because I am confused why the setup code is moved to a function that, by name, I assume to be called when wakeup happens (which is actually the check function)